storage/engine/mvcc_test.go

```diff
@@ -1236,6 +1236,39 @@ func TestMVCCResolveTxn(t *testing.T) {
 	}
 }
 
+// TestMVCCConditionalPutOldTimestamp tests a case where a conditional
+// put with an older timestamp happens after a put with a newer timestamp.
```

Comment:
explain what the test is for: That the conditional put uses the actual value at the timestamp as the basis for comparison first, and then may fail later with a WriteIntentError if that timestamp isn't recent.